src/ir/stack-utils.cpp

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "stack-utils.h"
+
+namespace wasm {
+
+namespace StackUtils {
+
+void compact(Block* block) {
```

Comment:
Yeah, `removeNops` would be a fine name here. This is used in Poppy IR passes that remove instructions. To avoid quadratic behavior, they transform the removed instructions into nops, then remove all the nops linearly at the end. So it doesn't need to do anything besides remove nops.